scripts/fetch_cloud_fraction.py

```
@@ -189,7 +189,24 @@ async def fetch_cloud_day(session: aiohttp.ClientSession, date: datetime) -> lis
         try:
             status, text = await earthdata_fetch(session, url, timeout=TIMEOUT)
             if status == 200:
-                if not text or "<html" in text[:200].lower():
+                if not text:
+                    diag_count = getattr(fetch_cloud_day, "_diag_silent", 0)
+                    if diag_count < 5:
+                        logger.warning(
+                            "Cloud %s: HTTP 200 empty body (silent failure)",
+                            date_str,
+                        )
+                        setattr(fetch_cloud_day, "_diag_silent", diag_count + 1)
```

Ruff (0.15.12)

[warning] Do not call `setattr` with a constant attribute value. It is not any safer than normal property access.

Replace `setattr` with assignment

(B010)